### PR TITLE
4187 REPL may become slow after large data output

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
https://github.com/Microsoft/RTVS/issues/4187

Map caret down and use the active editor buffer instead which is  just a few lines typically and does not include all the output.

### CODE REVIEW

File: src/R.sln [hunk 1]
```diff
@@ -130,6 +130,9 @@ EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.R.Containers.Windows.Test", "Windows\Containers\Test\Microsoft.R.Containers.Windows.Test.csproj", "{62E1CA8E-8398-46FC-B14F-DE0C5759DD2D}"
 EndProject
 Global
+	GlobalSection(Performance) = preSolution
+		HasPerformanceSessions = true
```
Comment (AlexanderSher) [inline src/R.sln]: ?

File: src/R/Editor/Impl/Completions/RCompletionContextExtensions.cs [hunk 2]
```diff
@@ -30,12 +38,20 @@ public static bool IsCaretInNamespace(this IEditorView view, out bool tripleColo
         }
 
         public static bool IsCaretInLibraryStatement(this IRIntellisenseContext context)
-            => context.Session.View.IsCaretInLibraryStatement();
+            => context.Session.View.IsCaretInLibraryStatement(context.EditorBuffer);
```
Comment (AlexanderSher) [inline src/R/Editor/Impl/Completions/RCompletionContextExtensions.cs]: Is it possible that `EditorBuffer` is null?

Comment (MikhailArkhipov) [reply src/R/Editor/Impl/Completions/RCompletionContextExtensions.cs]: Nope.

File: src/R.sln (hunk 1 above)
Comment (MikhailArkhipov) [reply src/R.sln]: Oh boy, VS adds this when you use profiler... Its harmless, but lemme try and undo it.

**Pull request merged**